### PR TITLE
Service requests workflow

### PR DESCRIPTION
**Pull request by christophermmeyer**
There are two workflows contained that integrate with ServiceNow. One creates a Change Request that requests a new host record to be created. The other approves Change Requests made via the other workflow. The approval workflow approves the ticket and then creates the host record within Address Manager.

### CODE REVIEW

**Comment by xiax** [comment]
The file Community/Service Requests/.DS_Store does not seem to be necessary for the workflow.

File: Community/Service Requests/service_request_host/js/service_request_host_page.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// limitations under the License.
+//
+// By: BlueCat Networks
+// Date: 04-05-18
+// Gateway Version: 18.6.1
+// Description: Example Gateway workflows
+
+// JavaScript for your page goes in here.
+
+function updateDeployStatus(deploy_token) {
```
Comment (xiax) [inline Community/Service Requests/service_request_host/js/service_request_host_page.js]: Hey I don't understand what this deployment status is for here. It doesn't seem to be part of the main logic of the workflow at all. Can you clarify why this and the server side endpoint are part of this workflow? Thanks.

File: Community/Service Requests/service_request_host/service_request_host_page.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    # Put form processing code here
+    flash('Successfully created ServiceNow Ticket: ' + response.json()['result']['number'], 'succeed')
+    flash('')
+    return redirect(url_for('service_request_hostservice_request_host_page'))
+
+
+@route(app, '/service_request_host/get_deploy_status', methods=['POST'])
+@util.rest_workflow_permission_required('service_request_host_page')
+@util.rest_exception_catcher
+def get_deploy_status():
```
Comment (xiax) [inline Community/Service Requests/service_request_host/service_request_host_page.py]: I'm not sure why this endpoint is necessary for this workflow.

File: Community/Service Requests/service_request_host/templates/service_request_host_page.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{{ text['title'] }}
+{% endblock %}
+
+{% block custom %}
+    <div id="deploy_status"></div>
+    <p>{{ text['info'] }}</p>
+    {% from "form_helper.html" import render_form %}
+    {{ render_form(form, action=url_for('service_request_hostservice_request_host_page_form'), id="service_request_host_form") }}
+
+    <input type="hidden" id="status_token" value="{{ status_token }}">
```
Comment (xiax) [inline Community/Service Requests/service_request_host/templates/service_request_host_page.html]: I don't see anything in the workflow that sets the status_token when rendering the html template.

File: Community/Service Requests/service_request_host/js/service_request_host_page.js (hunk 1 above)
Comment (christophermmeyer) [reply Community/Service Requests/service_request_host/js/service_request_host_page.js]: That was probably left over from a previous workflow I based this off of and I just missed it. I will remove this and recommit it.

File: Community/Service Requests/service_request_host/templates/service_request_host_page.html (hunk 3 above)
Comment (christophermmeyer) [reply Community/Service Requests/service_request_host/templates/service_request_host_page.html]: This one I'm not really sure about. I took a look at my local files and it's not there. So either I caught it and it never got updated, or something very strange happened. Either way, I got it taken care of!

**Comment by christophermmeyer** [comment]
@xiax Should be all set now!

File: Community/Service Requests/manage_service_requests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2018 BlueCat Networks. All rights reserved.
```
Comment (delmeherbert) [inline Community/Service Requests/manage_service_requests/__init__.py]: Please change date to 2019. Also in manage_service_requests_page.css, manage_service_requests_page.js, manage_service_requests_form.py, manage_service_requests_page.html, __init__.py, service_request_host_page.css, service_request_host_page.py, and service_request_host_page.html.

Comment (christophermmeyer) [reply Community/Service Requests/manage_service_requests/__init__.py]: Whoops, missed the 2018 ones! Should be covered on 2018 and 2017 now.

File: Community/Service Requests/service_request_host/js/service_request_host_page.js (hunk 1 above)
Comment (delmeherbert) [reply Community/Service Requests/service_request_host/js/service_request_host_page.js]: Hi Chris, looks like updateDeployStatus is still there.

File: Community/Service Requests/service_request_host/service_request_host_page.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
+# Copyright 2018 BlueCat Networks (USA) Inc. and its affiliates
```
Comment (delmeherbert) [inline Community/Service Requests/service_request_host/service_request_host_page.py]: Please change date to 2019. There are a few instances of 2018 and one of 2017 in the files.

Comment (christophermmeyer) [reply Community/Service Requests/service_request_host/service_request_host_page.py]: @delmeherbert Can you take a look at the latest files? That should have been changed with my last set of commits.

File: Community/Service Requests/service_request_host/js/service_request_host_page.js (hunk 1 above)
Comment (delmeherbert) [reply Community/Service Requests/service_request_host/js/service_request_host_page.js]: My apologies Chris, I looked at these files and not your very latest commits.

**Pull request merged**